meson_options.txt

```diff
@@ -0,0 +1,6 @@
+option('ipopt_dir', type: 'string', value: '',
+        description: 'Top-level dir for ipopt')
+option('incdir_numpy', type: 'string', value: '',
```

Comment:
Yeah they are. At least for cross-compilation, which is necessary for osx-arm64 builds on conda-forge. Cross-compilation took some time to get working so I'd really like if this could stay in! It doesn't really require any extra maintenance on your end but the option is really nice on my end where cross-compilation would be super helpful for our CI.

Reply:
Okay, as long as we have some sort of documentation (maybe just as comments here, e.g. what would be typical values passed in) I'm fine with keeping those. I don't see us touching these options so I want to at least get an idea of how to use them.

Reply:
> Finally, another use case of python_target/python_command that may be of more interest to you. Meson has Python as a dependency and by default, py3 = py_mod.get_installation() grabs the Python version used to run Meson. So consider the case that you use meson installed on your system using apt for example, but you want to build and install pyoptsparse for the Python version in your pyenv. Without separating python_target/python_command, this would not be possible because py3 = py_mod.get_installation() would grab your system installation, not what's in pyenv. By separating them you can now do py3_command = py_mod.get_installation() for getting the system installation associated with meson and py3_target = py_mod.get_installation('/path/to/pyenv/bin/python') to install to your pyenv installation.

This part of my previous comment answers your question about `python_target` I think. But I will add some notes in comments later and resolve this once I do